test/membrane_http_adaptive_stream/integration_test/sink_bin_integration_test.exs

```diff
@@ -0,0 +1,189 @@
+defmodule Membrane.HTTPAdaptiveStream.SinkBinIntegrationTest do
+  use ExUnit.Case, async: true
+
+  import Membrane.Testing.Assertions
+  alias Membrane.Testing
+
+  @doc """
+  The boolean flag below controls whether reference HLS content in fixtures directory will be created simultaneously with test content.
+  It should be set only when developing new HLS features that are expected to introduce changes to reference HLS files. Nevertheless it should
+  be done only locally to create and push new reference HLS files and this flag must not be set in remote repository. There is unit test in code below
+  that will cause CI to fail if this flag happens to be set on remote repository.
+  """
+  @create_fixtures false
+
+  @single_video_track_source [
+    {"https://raw.githubusercontent.com/membraneframework/static/gh-pages/samples/ffmpeg-testsrc.h264",
+     :H264, "single_video_track"}
+  ]
+  @single_video_track_ref_path "./test/membrane_http_adaptive_stream/integration_test/fixtures/single_video_track/"
+  @single_video_track_test_path "./test/membrane_http_adaptive_stream/integration_test/tmp/single_video_track/"
+
+  @audio_video_tracks_sources [
+    {"https://raw.githubusercontent.com/membraneframework/static/gh-pages/samples/test-audio.aac",
+     :AAC, "audio_track"},
+    {"https://raw.githubusercontent.com/membraneframework/static/gh-pages/samples/ffmpeg-testsrc.h264",
+     :H264, "video_track"}
+  ]
+  @audio_video_tracks_ref_path "./test/membrane_http_adaptive_stream/integration_test/fixtures/audio_video_tracks/"
+  @audio_video_tracks_test_path "./test/membrane_http_adaptive_stream/integration_test/tmp/audio_video_tracks/"
+
+  @audio_multiple_video_tracks_sources [
+    {"https://raw.githubusercontent.com/membraneframework/static/gh-pages/samples/big-buck-bunny/bun33s.aac",
+     :AAC, "audio_track"},
+    {"https://raw.githubusercontent.com/membraneframework/static/gh-pages/samples/big-buck-bunny/bun33s_480x270.h264",
+     :H264, "video_480x270"},
+    {"https://raw.githubusercontent.com/membraneframework/static/gh-pages/samples/big-buck-bunny/bun33s_540x360.h264",
+     :H264, "video_540x360"},
+    {"https://raw.githubusercontent.com/membraneframework/static/gh-pages/samples/big-buck-bunny/bun33s_720x480.h264",
+     :H264, "video_720x480"}
+  ]
+  @audio_multiple_video_tracks_ref_path "./test/membrane_http_adaptive_stream/integration_test/fixtures/audio_multiple_video_tracks/"
+  @audio_multiple_video_tracks_test_path "./test/membrane_http_adaptive_stream/integration_test/tmp/audio_multiple_video_tracks/"
+
+  defmodule TestPipeline do
+    use Membrane.Pipeline
+    alias Membrane.HTTPAdaptiveStream.{SinkBin, HLS}
+    alias Membrane.HTTPAdaptiveStream.Storages.FileStorage
+
+    @impl true
+    def handle_init(%{:sources => sources, :output_dir => output_dir}) do
+      sink_bin = %SinkBin{
```

Comment:
nit: don't you think that `HTTPAdaptiveStream.SinkBin` would be a bit more readable?